# Marko Melishchuk Notebook Big Ideas Final Project

In [10]:
!git clone https://github.com/oliviat10/Final-Project-MOK.git

Cloning into 'Final-Project-MOK'...
remote: Enumerating objects: 17, done.
remote: Counting objects: 100% (17/17), done.
remote: Compressing objects: 100% (14/14), done.
remote: Total 17 (delta 3), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (17/17), done.


Marko's Notebook

In [2]:
import numpy as np
import pandas as pd


air = pd.read_csv('toxicair.csv')


In [3]:
air.head()

TRI_FACILITY_ID   DOC_CTRL_NUM       FACILITY_NAME      STREET_ADDRESS  \
0  15025SSCLR400ST  1396100016272  USS-CLAIRTON PLANT  400 STATE ST MS 71   
1  15025SSCLR400ST  1396100016272  USS-CLAIRTON PLANT  400 STATE ST MS 71   
2  15106THHMM235HO  1387010137127         O HOMMEL CO         235 HOPE ST   
3  15106THHMM235HO  1387010137127         O HOMMEL CO         235 HOPE ST   
4  15025SSCLR400ST  1392065615205  USS-CLAIRTON PLANT  400 STATE ST MS 71   

  CITY_NAME COUNTY_NAME  STATE_COUNTY_FIPS_CODE STATE_ABBR  ZIP_CODE  REGION  \
0  CLAIRTON   ALLEGHENY                   42003         PA     15025       3   
1  CLAIRTON   ALLEGHENY                   42003         PA     15025       3   
2  CARNEGIE   ALLEGHENY                   42003         PA     15106       3   
3  CARNEGIE   ALLEGHENY                   42003         PA     15106       3   
4  CLAIRTON   ALLEGHENY                   42003         PA     15025       3   

   ...  NAICS_CODES INDUSTRY_CODE    SRS_ID PFAS_IND ENVIRONMENTAL_MEDIUM  \
0  ...          NaN           331   14183.0        0              AIR FUG   
1  ...          NaN           331   14183.0        0            AIR STACK   
2  ...          NaN           325  149583.0        0              AIR FUG   
3  ...          NaN           325  149583.0        0            AIR STACK   
4  ...          NaN           331    5090.0        0              AIR FUG   

  RELEASE_RANGE_CODE  TOTAL_RELEASE  REL_EST_AMT  RELEASE_NA  \
0                NaN          380.0        380.0           0   
1                NaN            0.0          0.0           0   
2                NaN            NaN          0.0           0   
3                NaN         1784.0       1784.0           0   
4                NaN        16000.0      16000.0           0   

  RELEASE_BASIS_EST_CODE  
0                      O  
1                      O  
2                      O  
3                      O  
4                      E  

[5 rows x 38 columns]

In [4]:
a_release = air['TOTAL_RELEASE'].sum()
print(a_release)

136976755.5773927


In [5]:
air.shape

(25536, 38)

The total amount (in pounds) of the toxic chemical released to air, water, land, and underground injection wells during the calendar year (January 1 - December 31). This means that there are 136976755.5773927 lbs of toxic chemicals released into the air (so far).

In [6]:
air['CHEM_NAME'].unique()

array(['Biphenyl', 'Lead', 'Ethylene', 'Naphthalene', 'Methanol',
       'Cresol (mixed isomers)', 'Ethyl acrylate', 'Toluene',
       'Cyanide compounds', 'Acetone', 'p-Cresol', 'Anthracene',
       'Barium compounds (except for barium sulfate (CAS No. 7727-43-7))',
       'Acrylic acid', 'Titanium tetrachloride', '1,2,4-Trimethylbenzene',
       'Phthalic anhydride', 'Maleic anhydride', 'Dibenzofuran',
       'Sodium hydroxide (solution)',
       'Nitrate compounds (water dissociable; reportable only when in aqueous solution)',
       'Ethylene glycol', 'Xylene (mixed isomers)',
       'Sulfuric acid (acid aerosols including mists, vapors, gas, fog, and other airborne forms of any particle size)',
       'Phosphoric acid', 'Zinc compounds',
       'Chromium compounds (except for chromite ore mined in the Transvaal Region)',
       'Nickel compounds', 'Certain glycol ethers', 'n-Butyl alcohol',
       'Butyl acrylate', 'p-Xylene', 'Ethylbenzene',
       'Molybdenum trioxide', 'Copper 

In [7]:
len(air['CHEM_NAME'].unique())

175

There are 175 different chemicals or chemical compounds being released into the air currently

In [9]:
(air['CITY_NAME'].unique()

array(['CLAIRTON', 'CARNEGIE', 'PITTSBURGH', 'SPRINGDALE', 'CHESWICK',
       'BRADDOCK', 'GLENSHAW', 'BRIDGEVILLE', 'BAKERSTOWN',
       'WEST MIFFLIN', 'WEST ELIZABETH', 'OAKMONT', 'BRACKENRIDGE',
       'CREIGHTON', 'MUNHALL', 'LEETSDALE', 'SEWICKLEY', 'VERONA',
       'HARWICK', 'MC KEES ROCKS', 'HOMESTEAD', 'WILMERDING',
       'CORAOPOLIS', 'NATRONA HEIGHTS', 'MONROEVILLE', 'MC KEESPORT',
       'GIBSONIA', 'BUNOLA', 'OAKDALE', 'CLINTON', 'JEFFERSON HILLS',
       'MC DONALD', 'TURTLE CREEK', 'INDIANOLA', 'MCKEESPORT', 'DUQUESNE',
       'NEVILLE ISLAND', 'SOUTH PARK', 'GLASSPORT', 'HEIDELBERG',
       'RANKIN', 'SHARPSBURG', 'MCKEES ROCKS', 'EAST PITTSBURGH',
       'LIBERTY BOROUGH', 'BLAWNOX', 'TARENTUM', 'DRAVOSBURG',
       'MONONGAHELA'], dtype=object)